components/sync15/ios/RustSyncTelemetryPing.swift

```diff
@@ -0,0 +1,622 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+public class RustSyncTelemetryPing {
+    public var version: Int
+    public var uid: String
+    public var events: [EventInfo]
+    public var syncs: [SyncInfo]
+
+    private static let EMPTY_UID = String(repeating: "0", count: 32)
+
+    init(version: Int, uid: String, events: [EventInfo], syncs: [SyncInfo]) {
+        self.version = version
+        self.uid = uid
+        self.events = events
+        self.syncs = syncs
+    }
+
+    static func empty() -> RustSyncTelemetryPing {
+        return RustSyncTelemetryPing(version: 1,
+                                     uid: EMPTY_UID,
+                                     events: [EventInfo](),
+                                     syncs: [SyncInfo]())
+    }
+
+    static func fromJSON(jsonObject: [String: Any]) throws -> RustSyncTelemetryPing {
+        guard let version = jsonObject["version"] as? Int else {
+            throw TelemetryJSONError.intValueNotFound(
+                message: "RustSyncTelemetryPing `version` property not found")
+        }
+
+        let events = unwrapFromJSON(jsonObject: jsonObject) { obj in
+            try EventInfo.fromJSONArray(
+                jsonArray: obj["events"] as? [[String: Any]] ?? [[String: Any]]())
+        } ?? [EventInfo]()
+
+        let syncs = unwrapFromJSON(jsonObject: jsonObject) { obj in
+            try SyncInfo.fromJSONArray(
+                jsonArray: obj["syncs"] as? [[String: Any]] ?? [[String: Any]]())
+        } ?? [SyncInfo]()
+
+        return RustSyncTelemetryPing(version: version,
+                                     uid: try stringOrNull(jsonObject: jsonObject,
+                                                           key: "uid") ?? EMPTY_UID,
+                                     events: events, syncs: syncs)
+    }
+
+    public static func fromJSONString(jsonObjectText: String) throws -> RustSyncTelemetryPing {
+        guard let data = jsonObjectText.data(using: .utf8) else {
+            throw TelemetryJSONError.invalidJSONString
+        }
+
+        let jsonObject = try JSONSerialization.jsonObject(with: data) as? [String: Any] ?? [String: Any]()
+        return try fromJSON(jsonObject: jsonObject)
+    }
+
+    func toJSON() throws -> [String: Any] {
```

Comment:
hmm do we need all the `toJSON` functions? do we expect this type to also be serializable?

Reply:
Yeah this and the `let` comment are fair points. I duplicated this code from `SyncTelemetryPing.swift` knowing that I needed most of it and adding the rest while I was there "just in case". But since we ultimately want to get rid of this serialization/deserialization logic and handle telemetry within a-s, I think stripping this file down to the bare essentials and treating it as a carrier of deserialized sync telemetry data makes a lot of sense.